File: ext/phalconry/http/client/adapter/adapterinterface.zep.c

#ifdef HAVE_CONFIG_H
#include "../../../../ext_config.h"
#endif

#include <php.h>
#include "../../../../php_ext.h"
#include "../../../../ext.h"

#include <Zend/zend_exceptions.h>

#include "kernel/main.h"


ZEPHIR_INIT_CLASS(Phalconry_Http_Client_Adapter_AdapterInterface) {

	ZEPHIR_REGISTER_INTERFACE(Phalconry\\Http\\Client\\Adapter, AdapterInterface, phalconry, http_client_adapter_adapterinterface, phalconry_http_client_adapter_adapterinterface_method_entry);

	return SUCCESS;

}

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, setOption);

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, setOptions);

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, setTimeout);

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, setProxy);

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, get);

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, head);

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, delete);

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, post);

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, put);

ZEPHIR_DOC_METHOD(Phalconry_Http_Client_Adapter_AdapterInterface, __invoke);

File: ide/0.4.6/Phalconry/http/client/adapter/AdapterInterface.zep.php
<?php

namespace Phalconry\Http\Client\Adapter;


interface AdapterInterface
{

    /**
     * @param mixed $option 
     * @param mixed $value 
     */
    public function setOption($option, $value);

    /**
     * @param mixed $options 
     */
    public function setOptions($options);

    /**
     * @param mixed $timeout 
     */
    public function setTimeout($timeout);

    /**
     * @param mixed $host 
     * @param mixed $port 
     * @param mixed $user 
     * @param mixed $pass 
     */
    public function setProxy($host, $port = 8080, $user = null, $pass = null);

    /**
     * @param mixed $request 
     * @param mixed $params 
     */
    public function get(\Phalconry\Http\Client\Request $request, $params = array());

    /**
     * @param mixed $request 
     * @param mixed $params 
     */
    public function head(\Phalconry\Http\Client\Request $request, $params = array());

    /**
     * @param mixed $request 
     * @param mixed $params 
     */
    public function delete(\Phalconry\Http\Client\Request $request, $params = array());

    /**
     * @param mixed $request 
     * @param mixed $params 
     */
    public function post(\Phalconry\Http\Client\Request $request, $params = array());

    /**
     * @param mixed $request 
     * @param mixed $params 
     */
    public function put(\Phalconry\Http\Client\Request $request, $params = array());

    /**
     * @param mixed $request 
     */
    public function __invoke(\Phalconry\Http\Client\Request $request);

}
